naucse/routes.py

```diff
@@ -211,3 +212,39 @@ def lesson(lesson, page):
     page = lesson.pages[page]
     g.vars = dict(page.vars)
     return render_page(page=page, page_wip=True)
+
+
+####### SESSION
+
+
+def session_template_or_404(run, session, page):
+    env = app.jinja_env.overlay(loader=session_template_loader)
+    run_link = str(run).split(" - ")[0]
```

Comment:
Don't parse the string representation of objects – what if someone decides to make it more useful and puts more info into it?
Use the individual properties. (Look into `Run.__str__` for their names.)